dashboard-widgets.el

```diff
@@ -98,6 +102,53 @@ preserved."
   :type 'integer
   :group 'dashboard)
 
+(defcustom dashboard-icon-type (or (require 'nerd-icons nil t)
+                                   (require 'all-the-icons nil t))
```

Comment:
I think it's good to prevent require in the root level. Let's prevent it by making sure the `dashboard-set-heading-icons` is non-nil:

```elisp
(and dashboard-set-heading-icons
    (or (require 'nerd-icons nil t)
        (require 'all-the-icons nil t)))
```

Make sure to place `dashboard-set-heading-icons` to the front to avoid compile error.